### PR TITLE
A Mime's Vow to Silence Cannot be Repaired Once Broken. (Mimes can Write Under the Vow Again.)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Breaking your Vow is now a permanent action. Here's the changes that come with that:

- Renamed "Speech" to "Break Vow" and changed description to match.
- You no longer get a permanent bad moodlet, just a temporary, lighter one. I calculated that breaking your vow of silence provides roughly the same emotional damage as getting your eyes stabbed out.
- Breaking vow now comes with a confirm alert.
- Mimes can once again write while under the vow.

## Why It's Good For The Game

After all this recent hullabaloo about mimes breaking and repairing their vow, and what the moodlets for that action should be, it was decided by a headcoder majority that mime vow breaking should be a permanent action.

image

But beyond headcoder decisions, I myself agree with it being permanent. Making vow breaking a temporary action that can be repaired after only 5 minutes makes ignoring the main gimmick of mime not only viable but essentially expected. When silence can usually cause moments where you NEED to explain yourself or else risk possibly dying, we're essentially pushing players to break a vow. Now, it will be seen as less of a potential option, because you can't undo it after 5 minutes.

## Changelog
:cl:
qol: Added a confirmation prompt to breaking your vow of silence.
balance: Mimes can no longer repair the vow of silence once broken.
balance: The negative moodlet from breaking your vow of silence is now lighter, and no longer permanent.
balance: Mimes are back to being able to write while under a vow of silence.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
this needs to reopen the mime job slot.

**Comment by optimumtact** [comment]
also while you are here, please reallow mimes to write.

**Comment by tralezab** [comment]
👍 and 👍, I'm on it.

File: code/modules/spells/spell_types/self/mime_vow.dm [hunk 1]
```diff
@@ -20,18 +19,18 @@
 /datum/action/cooldown/spell/vow_of_silence/Remove(mob/living/remove_from)
 	. = ..()
 	REMOVE_TRAIT(remove_from, TRAIT_MIMING, "[type]")
-	remove_from.clear_mood_event("vow")
+
+/datum/action/cooldown/spell/vow_of_silence/before_cast(atom/cast_on)
+	if(tgui_alert(usr, "Are you sure? There's no going back.", "Break Vow", list("I'm Sure", "Abort")) == "Abort")
```
Comment (JohnFulpWillard) [inline code/modules/spells/spell_types/self/mime_vow.dm:24]:
```suggestion
	if(tgui_alert(usr, "Are you sure? There's no going back.", "Break Vow", list("I'm Sure", "Abort")) != "I'm Sure")
```

**Comment by DATA-xPUNGED** [comment]
> this needs to reopen the mime job slot.

This should be the hop's job, don't make it automagically do that

**Comment by Namwty** [comment]
> This should be the hop's job, don't make it automagically do that

Does make sense, but at the same time has issues.

What if there isn't HoP or how is HoP even meant to know that mime broke vow ( or if they would even give a shit about it or know about it being a mechanic ). Mime is limited slot job as it is, and some people want to play it ( Lord have mercy on their souls ), breaking vow basically eliminates mime as a thing for round, so it makes sense for slot increase.

But at the same time, job slot opening is basically dead giveaway for mime breaking vow, which can hinder mime if they have gimmick in mind of some sorts.

Tbh maybe I just don't have faith in HoPs doing something like that, so maybe it's just me issue.


Regardless, people should just roundstart hunt down mimes and cut their tongue out as a safety measure, just stating facts... it's for greater good.

**Comment by tralezab** [comment]
Respectfully, yah honor, whatever happens to a mime after they start speakin is nan a my business...

**Comment by MrMelbert** [comment]
HoP job content

**Comment by optimumtact** [comment]
WHen and if the admin team show any drive to hold HoP's (or any heads) to doing their job i will consider your argument, but not before then.

**Comment by wesoda25** [comment]
Have you put thought into the case of antag mimes whose kit contains spoken spells?

**Comment by xzero314** [comment]
> WHen and if the admin team show any drive to hold HoP's (or any heads) to doing their job i will consider your argument, but not before then.

Mime heretics need to break their vow to cast spells. (Presumably cultist ones do too) wouldn't this be unfair to the mime that rolled those antagonists to have a way to 100% tell they are an antag that wants to keep quite?

**Comment by tralezab** [comment]
The solution to that is letting mimes bypass invocation requirements when playing certain antags. Mime shouldn't have to break Vow to play the antag they rolled.

**Comment by MrMelbert** [comment]
Disagree, the invocation is both one of the major tell of a spell and one of the ways to counter a spell (mute toxin). I already snowflake allowed mimes to cast Mansus Grasp and I kinda regret it

**Comment by tralezab** [comment]
Is there an option I'm not thinking of that keeps the invocation and the vow?

**Comment by Rex9001** [comment]
> Is there an option I'm not thinking of that keeps the invocation and the vow?

Maybe some sort of emote invocation? idk

**Comment by Fikou** [comment]
> WHen and if the admin team show any drive to hold HoP's (or any heads) to doing their job i will consider your argument, but not before then.

we did this term though

**Comment by DATA-xPUNGED** [comment]
> > WHen and if the admin team show any drive to hold HoP's (or any heads) to doing their job i will consider your argument, but not before then.
> 
> Mime heretics need to break their vow to cast spells. (Presumably cultist ones do too) wouldn't this be unfair to the mime that rolled those antagonists to have a way to 100% tell they are an antag that wants to keep quite?

Cultist ones don't, if i remember correctly. They can actually speak in the cult common without breaking the vow

**Comment by DaydreamIQ** [comment]
So this might be a complete non issue if I'm recalling this improperly but what if someone gets the mimery book from a traitor surplus crate? Are they just shit outta luck with an item that doesn't work or are they forced to commit to the vow forever?

**Comment by tralezab** [comment]
You either get the vow and the spells or neither.

**Comment by ATH1909** [comment]
> Is there an option I'm not thinking of that keeps the invocation and the vow?

maybe some way to fake/unfake a vow at-will if you're an antag mime?

**Comment by tralezab** [comment]
> > Is there an option I'm not thinking of that keeps the invocation and the vow?
> 
> maybe some way to fake/unfake a vow at-will if you're an antag mime?

Like not speaking as if you had the vow? You can do that at any time.

**Comment by Jezzared** [comment]
> Like not speaking as if you had the vow? You can do that at any time.

Breaking makes your emoted no longer silent. Things like "gasps silently" or "mimes an exaggerated cough" will just show up as "gasps" and "coughs", which can be a very large tell that a mime is up to no good.

**Comment by CGoyen** [comment]
This is a bad solution to a simple problem.

**Comment by t3hSurge** [comment]
I hugely agree with CGoyen.  Mending your vow took forever anyway, but it's deeply useful in antagonistic and RP scenarios.  So with this change, if I'm a syndicate mime and get advanced mimery I can't do any "evil talking" IC to further my goals, or work with somebody?

This is the opposite of what I want to see out of the mime.  I'm not sure what the recent hullabaloo you refer to is, but maybe the valid status of talking mimes needs to be made clearer to the playerbase or something.  I actually never noticed any moodlet issues in my mime rounds where I had to break vow, but again that was usually for traitoring so I'm pretty sure you get a big buff anyway.  I hate mimes that use paper gimmick-less.  The mime's crayon or some ascii art is all I ever do with paper.

Basically, instead of moving the mime towards more gesticulation and RP, a feature was replaced with a soft-suicide button.  Bad solution.

**Comment by Jezzared** [comment]
Don't Make Me Tap The Sign.

**Comment by optimumtact** [comment]
what he said

**Comment by CGoyen** [comment]
The mime has a history with the ventriloquist dummy as well. Even starts with one near by. If you break the vow to use it, then you can not use any other mime abilities.

If the issue was people breaking their vow and having to wait 5 minutes, with the penalty of being valid for the entire station, maybe extend the time they have to wait based on how much they have talked.

**Comment by t3hSurge** [comment]
I’d rather see speaking mime codified as a capital crime under the captain’s discretion like any other. Silenced medically, borgificaciously, imprisoned or terminated. That or just remove the feature altogether and make mutism permanent like clown’s clumsiness.
